recipe/meta.yaml

```diff
@@ -1,19 +1,22 @@
 {% set name = "configurable-http-proxy" %}
 {% set version = "3.1.0" %}
 {% set node_version = os.environ.get('NODEJS_VERSION') or '6.*' %}
+{% set node_major = node_version.split('.')[0] %}
+{% set build = 1 %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  fn: {{ version }}.tar.gz
+  fn: {{ name }}-{{ version }}.tar.gz
   url: https://registry.npmjs.org/{{ name }}/-/{{ name }}-{{ version }}.tgz
   sha256: eb41627ed15505261dbbb48eb4b9f1c6cd31a884b0f1afa9001a671b67262ad0
 
 build:
-  number: 0
-  script: npm install -g .
+  number: {{ build }}
+  string: "node{{node_major}}_{{ build}}"
+  script: npm pack; npm install -g {{name}}-{{version}}.tgz  # [not win]
```

Comment:
For multiline scripts, one can do this.

ref: https://github.com/conda-forge/waf-feedstock/blob/ad1e7be11615070c8940ff1d2ebc4d2cfd5ed408/recipe/meta.yaml#L15-L18